File: back/JwtLoginAPI/Domain/Entities/Pokemon.cs
﻿using JwtLoginAPI.Domain.Entities;

namespace JwtLoginAPI.Domain.Commands.Entities
{
    public class Pokemon
    {
        public int Id { get; set; }
        public string Name { get; set; } = String.Empty;
        public string Description { get; set; } = String.Empty;
        public string Category { get; set; } = String.Empty;
        public Gender Gender { get; set; }
        public double Height { get; set; }
        public double Weight { get; set; }
        public ICollection<string>? Type { get; set; }
        public ICollection<string>? Weaknesses { get; set; }
        public List<Ability>? Abilities { get; set; }
    }

    public enum Gender
    {
        both,
        masculine,
        female,
        unknown
    }

}

File: front/src/services/pokemon.js
import { http } from './config'

export default {
    create:(parans) => {
        var formData = new FormData()
        formData.append('Name', parans.Name)
        formData.append('Description', parans.Description)
        formData.append('Category', parans.Category)
        formData.append('Gender', parans.Gender)
        formData.append('Height', parans.Height)
        formData.append('Weight', parans.Weight)
        formData.append('Type', parans.Type)
        formData.append('Weaknesses', parans.Weaknesses)
        formData.append('Photo', parans.Photo)

        for (let i = 0; i < parans.Abilities.length; i++) {
            formData.append('Abilities[]', parans.Abilities[i]);
        }
        
        return http.post('api/pokedex/createpokemon', formData)
    },
}